tests/FakeItEasy.Specs/DummyCreationSpecs.cs

```diff
@@ -471,6 +472,95 @@ public void ValueTupleCreation((Foo foo, string s) dummy)
                 .x(() => dummy.s.Should().BeOfType<string>());
         }
 
+        [Scenario]
+        public void LazyCreation(Lazy<string> dummy)
+        {
+            "Given a type"
+                .See<string>();
+
+            "When a dummy lazy of that type is requested"
+                .x(() => dummy = this.CreateDummy<Lazy<string>>());
+
+            "Then it returns a lazy"
+                .x(() => dummy.Should().BeOfType<Lazy<string>>());
+
+            "And the lazy's value isn't created yet"
+                .x(() => dummy.IsValueCreated.Should().BeFalse());
+
+            "And the lazy's value returns a dummy of its type"
+                .x(() => dummy.Value.Should().BeOfType<string>());
+        }
+
+        [Scenario]
+        public void TaskCreation(Task dummy)
+        {
+            "Given the Task type"
+                .See<Task>();
+
+            "When a dummy task is requested"
+                .x(() => dummy = this.CreateDummy<Task>());
+
+            "Then it returns a task"
+                .x(() => dummy.Should().BeAssignableTo<Task>());
+
+            "And the task is completed successfully"
+                .x(() => dummy.Status.Should().Be(TaskStatus.RanToCompletion));
+        }
+
+        [Scenario]
+        public void TaskOfTCreation(Task<string> dummy)
+        {
+            "Given a type"
+                .See<string>();
+
+            "When a dummy task of that type is requested"
+                .x(() => dummy = this.CreateDummy<Task<string>>());
+
+            "Then it returns a task"
+                .x(() => dummy.Should().BeOfType<Task<string>>());
+
+            "And the task is completed successfully"
+                .x(() => dummy.Status.Should().Be(TaskStatus.RanToCompletion));
+
+            "And the task's result is a dummy of the value type"
+                .x(() => dummy.Result.Should().BeOfType<string>());
+        }

```

Comment:
`Task` and `Lazy` dummies were covered in other specs (`TaskSpecs` and `LazySpecs`), but in an indirect way. I thought it best to cover them in this file, along with other dummy creation specs.

Reply:
Feels slightly redundant, but it unlikely to cause problems. What do you think about absorbing the other specs into this?

Reply:
Well, there isn't much more is the existing specs. It's just that instead of using `A.Dummy<T>()`, they call unconfigured methods returning a T. They also test the case where T is not dummyable. If you want, I can add specs to cover the non-dummyable case, and remove LazySpecs and TaskSpecs.

Reply:
I sort of like that idea, if it's not too much trouble.

Reply:
Will do